rpc/lib/server/handlers.go

```diff
@@ -118,6 +118,7 @@ func makeJSONRPCHandler(funcMap map[string]*RPCFunc, logger log.Logger) http.Han
 		// The Server MUST NOT reply to a Notification, including those that are within a batch request.
 		if request.ID == "" {
 			logger.Debug("HTTPJSONRPC received a notification, skipping... (please send a non-empty ID if you want to call a method)")
+			WriteRPCResponseHTTP(w, types.RPCInvalidParamsError(request.ID, errors.New(`expecting "id"`)))
```

Comment:
this violates the JSONRPC specification: `The Server MUST NOT reply to a Notification, including those that are within a batch request.`

Reply:
Oh, TIL, thanks for that, I did not know about that, I've been in plain HTTP land for too long :)